components/org.wso2.carbon.identity.api.server.authenticators/org.wso2.carbon.identity.api.server.authenticators.v1/src/main/java/org/wso2/carbon/identity/api/server/authenticators/v1/core/ServerAuthenticatorManagementService.java

```diff
@@ -511,6 +512,9 @@ private Authenticator addLocalAuthenticator(LocalAuthenticatorConfig config) {
         authenticator.setName(config.getName());
         authenticator.setDisplayName(config.getDisplayName());
         authenticator.setIsEnabled(config.isEnabled());
+        if (config.getProperties().length > 0 && IS_API_BASED_SUPPORTED.equals(config.getProperties()[0].getName())) {
```

Comment:
shall we add a null check here is there is any possibility of config.getProperties() being null